pkg/kotsadmsnapshot/backup.go

```diff
@@ -930,3 +946,59 @@ func prepareIncludedNamespaces(namespaces []string) []string {
 	}
 	return includedNamespaces
 }
+
+// excludeShutdownPodsFromBackup will exclude pods that are in a shutdown state from the backup
+// this is to prevent the hook backup from failing if a pod is in a shutdown state and cannot be backed up
+func excludeShutdownPodsFromBackup(ctx context.Context, clientset kubernetes.Interface, backup *velerov1.Backup) (err error) {
+	namespaces := backup.Spec.IncludedNamespaces
+	if len(namespaces) == 0 {
+		return
+	}
+
+	// if the backup includes all namespaces, we need to get the list of namespaces
+	if namespaces[0] == "*" {
+		namespaces, err = getNamespaces(ctx, clientset)
+		if err != nil {
+			return errors.Wrap(err, "failed to get namespaces")
+		}
+	}
+
+	for _, namespace := range namespaces {
+		pods, err := clientset.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+			FieldSelector: "status.phase=Failed",
+		})
+		if err != nil {
+			return errors.Wrapf(err, "failed to list pods in namespace %s", namespace)
+		}
+
+		for _, pod := range pods.Items {
+			if pod.Status.Reason == "Shutdown" {
+				logger.Infof("Excluding pod %s in namespace %s from backup", pod.Name, namespace)
+				// add velero.io/exclude-from-backup=true label to pod
+				if pod.Labels == nil {
+					pod.Labels = map[string]string{}
+				}
+
+				pod.Labels["velero.io/exclude-from-backup"] = "true"
```

Comment:
they are constants already defined for the exclude key and value:
```suggestion
				pod.Labels[kotsadmtypes.ExcludeKey] = kotsadmtypes.ExcludeValue
```